### PR TITLE
added helm chart for running holmes as a deployment in the cluster

### PR DESCRIPTION
**Pull request by arikalon1**
pulling robusta store configuration, from the existing secret, if exists fix some small bugs

### CODE REVIEW

File: helm/holmes/templates/holmesgpt-service-account.yaml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      - configmaps
+      - daemonsets
+      - deployments
+      - events
+      - namespaces
+      - persistentvolumes
+      - persistentvolumeclaims
+      - pods
+      - pods/status
+      - pods/exec
```
Comment (aantn) [inline helm/holmes/templates/holmesgpt-service-account.yaml:23]: I would probably add a flag and make this optional (it's a higher permission than the rest and we can do without it for the default toolsets anyway)

File: holmes/core/tools.py [hunk 2]
```diff
@@ -157,6 +157,7 @@ def check_prerequisites(self):
             except subprocess.CalledProcessError as e:
                 self._enabled = False
                 self._disabled_reason = f"prereq check failed w/ errorcode {e.returncode}"
+                logging.exception(f"Toolset {self.name} : Failed to run prereq command {prereq}")
```
Comment (aantn) [inline holmes/core/tools.py:160]: Could you add a log for this elsewhere? We're deliberately not logging it here to keep the CLI output clean.

File: holmes/core/supabase_dal.py [hunk 3]
```diff
@@ -1,52 +1,108 @@
...
+            config = RobustaConfig(**yaml_content)
+            for conf in config.sinks_config:
+                if "robusta_sink" in conf.keys():
+                    token = conf["robusta_sink"].get("token")
+                    return RobustaToken(**json.loads(base64.b64decode(token)))
+
+        return None
+
+    def __init_config(self) -> bool:
+        robusta_token = self.__load_robusta_config()
```
Comment (aantn) [inline holmes/core/supabase_dal.py:65]: Could you document the behaviour somewhere - either in the helm values or in this file - that we load from robusta token on disk first if possible, and if not fallback to environment variables if given

File: helm/holmes/values.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+
+# Optional openai key and endpoint to create openai queries.
+openaiKey: ""
+openaiEndpoint: ""
+openaiType: ""
```
Comment (aantn) [inline helm/holmes/values.yaml:5]: What does openaiType refer to? Same question about version?

File: helm/holmes/values.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+openaiVersion: ""
+openaiDeployment: ""
+aiModel: ""
+
+certificate: "" # base64 encoded
```
Comment (aantn) [inline helm/holmes/values.yaml:10]: What is this used for?

File: helm/holmes/templates/holmes.yaml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+                key: openaiKey
+          - name: OPENAI_ENDPOINT
+            value: {{ .Values.openaiEndpoint }}
+          - name: OPENAI_TYPE
+            value: {{ .Values.openaiType }}
+          - name: OPENAI_VERSION
+            value: {{ .Values.openaiVersion }}
+          - name: OPENAI_DEPLOYMENT
+            value: {{ .Values.openaiDeployment }}
+          - name: AI_MODEL
```
Comment (aantn) [inline helm/holmes/templates/holmes.yaml:55]: I think the names you're using here don't all match up to the names read in Config.load_from_env. 
E.g. AI_MODEL should just be MODEL

File: helm/holmes/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v2
+name: holmes
+description: HolmesGPT Helm chart for Kubernetes
+type: application
+
+# these are set to the right value by .github/workflows/release.yaml
```
Comment (aantn) [inline helm/holmes/Chart.yaml:6]: We should do that!

**Review by aantn — APPROVED** [review]
Looks good, left some comments.

File: holmes/core/tools.py (hunk 2 above)
Comment (arikalon1) [reply holmes/core/tools.py:160]: fixed

File: holmes/core/supabase_dal.py (hunk 3 above)
Comment (arikalon1) [reply holmes/core/supabase_dal.py:65]: I'll put a comment here

File: helm/holmes/values.yaml (hunk 5 above)
Comment (arikalon1) [reply helm/holmes/values.yaml:10]: for self hosted environments, some users use self signed certificates.
They can add it here, to trust their self signed certs

File: helm/holmes/Chart.yaml (hunk 7 above)
Comment (arikalon1) [reply helm/holmes/Chart.yaml:6]: yeah, I know
We will do it soon

File: helm/holmes/values.yaml (hunk 4 above)
Comment (arikalon1) [reply helm/holmes/values.yaml:5]: removing for now, leaving only the openaikey

File: helm/holmes/templates/holmes.yaml (hunk 6 above)
Comment (arikalon1) [reply helm/holmes/templates/holmes.yaml:55]: removing

**Pull request closed without merging**